tests/unit/clients/workflow_test.py

```diff
@@ -108,287 +112,326 @@ def test_build_time_filter_dict_with_close_date_range(oldest_close_date, latest_
     assert time_filter_dict['closeTimeFilter'] == {'oldestDate': oldest_close_date, 'latestDate': latest_close_date}
 
 
-def test_count_open_workflow_executions_with_oldest_start_time(workflow_config, workflow_client, boto_client, oldest_start_date):
-    workflow_client.count_open_workflow_executions(
-        oldest_start_date=oldest_start_date,
-    )
-    boto_client.count_open_workflow_executions.assert_called_with(
-        domain=workflow_config.domain,
-        startTimeFilter=dict(oldestDate=oldest_start_date),
-    )
-
-
-def test_count_open_workflow_executions_with_start_time_range(
-        workflow_config,
-        workflow_client,
-        boto_client,
-        oldest_start_date,
-        latest_start_date
-):
-    workflow_client.count_open_workflow_executions(
-        oldest_start_date=oldest_start_date,
-        latest_start_date=latest_start_date,
-    )
-    boto_client.count_open_workflow_executions.assert_called_with(
-        domain=workflow_config.domain,
-        startTimeFilter=dict(
-            oldestDate=oldest_start_date,
-            latestDate=latest_start_date,
-        ),
-    )
+class TestCountOpenWorkflowExecutions:
+
+    @pytest.fixture
+    def mocked_boto_client(self):
+        mocked_boto_client = mock.Mock()
+        mocked_boto_client.count_open_workflow_executions.return_value = {
+            'count': 123,
+            'truncated': False
+        }
+        return mocked_boto_client
+
+    @pytest.fixture
+    def workflow_client(self, workflow_config, mocked_boto_client):
+        return WorkflowClient(workflow_config, mocked_boto_client)
+
+    def test_count_open_workflow_executions_with_oldest_start_time(
+            self,
+            workflow_config,
+            workflow_client,
+            mocked_boto_client,
+            oldest_start_date
+    ):
+        count = workflow_client.count_open_workflow_executions(
+            oldest_start_date=oldest_start_date,
+        ).count
+        assert count == 123
```

Comment:
You can perform a stronger assertion if you compared the return of this function with the object you assigned the mock to. That way you can assert both count and truncated